tasking/pulp/tasking/celery_app.py

```diff
@@ -69,6 +72,51 @@ def initialize_worker(sender, instance, **kwargs):
     # Delete any potential old state
     delete_worker(sender, normal_shutdown=True)
 
+    if sender.startswith(constants.RESOURCE_MANAGER_WORKER_NAME):
+        get_resource_manager_lock(sender)
+
+
+def get_resource_manager_lock(name):
+    """
+    Tries to acquire the resource manager lock. If the lock cannot be acquired immediately, it
+    will wait until the currently active instance becomes unavailable, at which point the worker
+    cleanup routine will clear the lock for us to acquire. A worker record will be created so that
+    the waiting resource manager will appear in the Status API. We override the SIGTERM signal
+    handler so that that the worker record will be immediately cleaned up if the process is killed
+    while in this states.
+
+    :param name:   The hostname of the worker
+    :type  name:   basestring
+    """
+    assert name.startswith(constants.RESOURCE_MANAGER_WORKER_NAME)
```

Comment:
I like putting assertions like this in code, but I haven't seen much of it in the pulp codebase before. :+1: